astropy/coordinates/baseframe.py

```diff
@@ -597,13 +597,8 @@ def __init__(self, *args, **kwargs):
                 'Coordinate frame got unexpected keywords: {0}'.format(
                     list(kwargs)))
 
-        self._data = representation_data
-
-        # We do ``is not None`` because self._data might evaluate to false for
-        # empty arrays or data == 0
-        if self._data is not None:
-            self._rep_cache = dict()
-            self._rep_cache[self._data.__class__.__name__, False] = self._data
+        self._rep_cache = dict()
```

Comment:
I think it should be in `update_representation_data`.  The purpose here is to cache other representations of the data that are requested, while always maintaining `self._data` as the original representation.  If `self._data` gets updated in place then this cache gets invalidated and should be cleared.